src/IntegratedExperts/BehatScreenshot/ScreenshotContext.php

```diff
@@ -74,6 +82,29 @@ public function __construct($parameters = [])
     }
 
     /**
+     * Init function before tests run.
+     *
+     * @param BeforeSuiteScope $scope
+     *
+     * @BeforeSuite
+     */
+    public static function init(BeforeSuiteScope $scope)
+    {
+        foreach ($scope->getSuite()->getSetting('contexts') as $context) {
+            if (isset($context['IntegratedExperts\BehatScreenshot\ScreenshotContext'][0])) {
+                $contextSettings = $context['IntegratedExperts\BehatScreenshot\ScreenshotContext'][0];
+            }
+        }
+
+        self::$purge = (getenv('BEHAT_SCREENSHOT_PURGE')) ? intval(getenv('BEHAT_SCREENSHOT_PURGE')) : (isset($contextSettings['purge'])) ? $contextSettings['purge'] : self::$purge;
+        $dir = (isset($contextSettings['dir'])) ? $contextSettings['dir'] : __DIR__.'/screenshot';
+
+        if (self::$purge === 1 && isset($dir)) {
```

Comment:
please never use `isset($dir)` for a variable value - the variable should be defined before used.